app/controllers/works_controller.rb

```diff
@@ -283,6 +284,14 @@ def create
     @work = Work.new(work_params)
     @chapter = @work.first_chapter
     @chapter.attributes = work_params[:chapter_attributes] if work_params[:chapter_attributes]
+    if work_params[:series_attributes]
```

Comment:
Is it possible to move most of this parsing logic into the work model, instead of having them in the work controller?

I think the setup can be similar to how we handle related works, with a method for holding the temporary state, both for the preview mode and for preserving info in case of form errors:

https://github.com/otwcode/otwarchive/blob/bedbbef8c43585d97b5a56062355921e4c8f5fa7/app/models/work.rb#L934-L935

and a method for actually saving the state:

https://github.com/otwcode/otwarchive/blob/bedbbef8c43585d97b5a56062355921e4c8f5fa7/app/models/work.rb#L973-L974

(It may not be 100% the case though, since related works don't show up in preview mode right now.)

Reply:
Hi! I'll take a look at it in a week or two if that's okay

Reply:
That's totally fine, no rush!

Reply:
About similarity with related works behavior: related works are still saved if we set it up in editor, then open work's preview and cancel the changes, so I guess there is the same problem that I'm trying to fix for series. But I removed parsing logic from controller and changed it a bit.

There are some other things I think worth noting.
It took some time for me to update the system (vagrant etc; I'm working on Windows) with the last version; don't know if it's only my problem, but just in case - I couldn't load the site from the host (got ERR_CONNECTION_REFUSED error) until added this setting to Vagrantfile:
config.vm.network "forwarded_port", guest: 7000, host: 7000, host_ip: "127.0.0.1"

Also, while working on this problem, I discovered another one; not sure if it isn't already in Jira, but anyway here are the steps to reproduce it:
1. Check "This work is part of a series" and set a new series title in editor
2. Go to Preview
3. On Preview page, click "Edit" button
4. Open Preview again
Actual result: new series with set title has been added to the database several times. You can see it in a dropdown list back on Edit page.
I wasn't sure about the correct behavior here: can there be different series with the same title? If not, then it is possible to just check series by title before saving like it works with id now. Or maybe series title is unique for one user.

Reply:
Sorry, apparently the last changes don't seem to work properly, and this problem with series is the result (like several other failed test cases). So for now I can't see how to manage passing the new series id other than the way it was done before, using @serial.

Reply:
I've added series id update for new series on saving - not sure if this way is better than using `@serial`; but there are still some tests failing:
features/other_b/series.feature:76
features/other_b/series.feature:225
They fail locally, too, but when I try to reproduce the scenario in an app, everything seems to work just fine.
I tried my best to investigate, but at this point I have no idea what can go wrong there; series list in the footer displays all the series and the one at the top ignores one, even though they use the same data, from what I can see.

Reply:
I think one of the issues here is that the new series shouldn't be saved until the work is saved. [If you delete the last work from a series, it deletes the series](https://github.com/otwcode/otwarchive/blob/d2568009c73f19c24248a07c9bc27717460766fa/app/models/serial_work.rb#L20-L25), so I think creating empty series in the first place could be considered a bug. The code surrounding parameters should get a bit simpler if you delete this line:
https://github.com/otwcode/otwarchive/blob/c08b2b6159e415ac0df7519bc33875ef1a0e8bdb/app/models/work.rb#L621

([There are some tests which rely on the existing behavior](https://github.com/otwcode/otwarchive/blob/1f7610f25b28388bd6b2beb4715f2a80ece78ef7/features/users/user_delete.feature#L98-L109), but I don't think it's a good plan to keep the buggy behavior around just for testing purposes. It'd probably be better to modify that scenario by creating a `{user} has an empty series` step that uses a factory to create the empty series.)

As for the preview issue, that looks like it's caused by a caching issue related to [AO3-3455](https://otwarchive.atlassian.net/browse/AO3-3455) and [AO3-5041](https://otwarchive.atlassian.net/browse/AO3-5041). Caching is disabled in the development environment, but not in tests. As long as red approves, it makes sense to me to add the `@disable_caching` tag to those tests and add this to AO3-5041 (not AO3-3455, that's a slightly more complex issue) once this PR gets merged.

Reply:
I checked locally - adding `@disable_caching` tag really helped with those tests!

Reply:
Yeah, it should be fine to use `@disable_caching` and add a `TODO: AO3-5041 etc.` to eventually remove the tag.